From: Jason Rhinelander <jason@imaginary.ca>
Date: Mon, 30 May 2022 20:23:25 -0300
Subject: Build SQLiteCpp statically

---
 external/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/external/CMakeLists.txt b/external/CMakeLists.txt
index 5ca5304..c7efd13 100644
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -169,4 +169,7 @@ set(SQLITECPP_RUN_DOXYGEN OFF CACHE BOOL "" FORCE)
 set(SQLITECPP_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
 set(SQLITECPP_BUILD_TESTS OFF CACHE BOOL "" FORCE)
 
+set(BUILD_SHARED_LIBS_SAVED "${BUILD_SHARED_LIBS}")
+set(BUILD_SHARED_LIBS OFF)
 add_subdirectory(SQLiteCpp)
+set(BUILD_SHARED_LIBS "${BUILD_SHARED_LIBS_SAVED}")
