From: Jason Rhinelander <jason@imaginary.ca>
Date: Mon, 30 May 2022 20:51:56 -0300
Subject: Allow older fmt

---
 external/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/external/CMakeLists.txt b/external/CMakeLists.txt
index c7efd13..3ebde44 100644
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -61,7 +61,7 @@ endmacro()
 
 system_or_submodule(OXENC oxenc liboxenc>=1.0.3 oxen-encoding)
 system_or_submodule(OXENMQ oxenmq liboxenmq>=1.2.12 oxen-mq)
-system_or_submodule(FMT fmt fmt>=8.0.0 fmt)
+system_or_submodule(FMT fmt fmt fmt)
 
 add_subdirectory(db_drivers)
 add_subdirectory(easylogging++ easyloggingpp)
